sky/skylet/providers/aws/config.py

```diff
@@ -1007,7 +1012,22 @@ def _update_inbound_rules(target_security_group, sgids, config):
     ip_permissions = _create_default_inbound_rules(
         sgids, user_specified_rules, extended_rules
     )
-    target_security_group.authorize_ingress(IpPermissions=ip_permissions)
+    try:
+        target_security_group.authorize_ingress(IpPermissions=ip_permissions)
+    except botocore.exceptions.ClientError as e:
+        if str(e) == _DUPLICATED_INBOUND_RULES_MSG:
+            logger.warning(
+                f"{cf.bold(target_security_group.group_name)} already has "
+                f"ingress rules matching {cf.bold(ip_permissions)}. "
+                "Skipping ingress rule update."
+            )
```

Comment:
Is it possible that only part of the `ip_permissions` duplicates? Will this makes the other `ip_permissions` failed to be updated in the security group?

Reply:
Great point! Thanks for catching that bug!

Reply:
sorry, i don't understand why the latest changes will help with the issue. If the `ip_permssion` is a range from `0-1000` and there is a existing `ip_permssion` for port `22`. Will the current implementation skip the update of `0-21` and `23-1000`?

The latest commit 5a65ae0 seems unnecessary as for none duplication error, it seems fine to fail early as the provisioning process will be interrupted anyway.

Reply:
The issue will only triggered when the inbound rule is exactly the same. It is fine to have some overlaps.
One test result:
<img width="949" alt="image" src="https://github.com/skypilot-org/skypilot/assets/74357442/83b42ddc-69cf-4406-9891-ec56b9b526a7">

Reply:
Re fail early: Good point! Done. Thanks!

Reply:
Same for Azure
<img width="1604" alt="image" src="https://github.com/skypilot-org/skypilot/assets/74357442/8c9cfe7d-c734-4ccc-82c4-b8bf21a0f535">